library/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/recorder/DebouncerTest.kt

```diff
@@ -0,0 +1,168 @@
+ */
+
+package com.datadog.android.sessionreplay.recorder
+
+import android.os.Handler
+import com.datadog.android.sessionreplay.utils.ForgeConfigurator
+import com.nhaarman.mockitokotlin2.any
+import com.nhaarman.mockitokotlin2.eq
+import com.nhaarman.mockitokotlin2.times
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.whenever
+import fr.xgouchet.elmyr.Forge
+import fr.xgouchet.elmyr.junit5.ForgeConfiguration
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.quality.Strictness
+import java.util.concurrent.TimeUnit
+import kotlin.math.ceil
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ForgeConfiguration(ForgeConfigurator::class)
+internal class DebouncerTest {
+
+    @Mock
+    lateinit var mockHandler: Handler
+
+    lateinit var testedDebouncer: Debouncer
+
+    @BeforeEach
+    fun `set up`() {
+        testedDebouncer = Debouncer(mockHandler, TEST_MAX_DELAY_THRESHOLD_IN_NS)
+    }
+
+    @Test
+    fun `M delegate to the delayed handler W debounce { first request }`() {
+        // Given
+        val fakeRunnable = TestRunnable()
+        whenever(mockHandler.postDelayed(any(), any())).then {
+            (it.arguments[0] as Runnable).run()
+            true
+        }
+
+        // When
+        testedDebouncer.debounce(fakeRunnable)
+
+        // Then
+        verify(mockHandler).removeCallbacksAndMessages(null)
+        verify(mockHandler).postDelayed(any(), eq(Debouncer.DEBOUNCE_TIME_IN_NS))
+    }
+
+    @Test
+    fun `M skip the handler and execute the runnable in place W debounce { threshold reached }`() {
+        // Given
+        val fakeRunnable = TestRunnable()
+        val fakeSecondRunnable = TestRunnable()
+        testedDebouncer.debounce(fakeRunnable)
+        Thread.sleep(TimeUnit.NANOSECONDS.toMillis(TEST_MAX_DELAY_THRESHOLD_IN_NS))
+
+        // When
+        testedDebouncer.debounce(fakeSecondRunnable)
+
+        // Then
+        verify(mockHandler, times(1)).postDelayed(
+            any(),
+            eq(Debouncer.DEBOUNCE_TIME_IN_NS)
+        )
+        verify(mockHandler, times(2)).removeCallbacksAndMessages(null)
+        assertThat(fakeRunnable.wasExecuted).isFalse
+        assertThat(fakeSecondRunnable.wasExecuted).isTrue
+    }
+
+    @Test
+    fun `M execute the runnable once W debounce { high frequency, delay threshold reached  }`(
+        forge: Forge
+    ) {
+        // Given
+        val fakeDelayedRunnables = forge.aList(size = forge.anInt(min = 0, max = 10)) { TestRunnable() }
```

Comment:
having `min=0` will make it possible to have empty `fakeDelayedRunnables` list, leading to the division by zero on the line below. Same for the test below.

Reply:
oh damn...yes, I actually had `min=1` there at some point I don't know why I changed it back. Nicely spotted.